pkg/wallet/wallet.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package wallet
+
+import (
+	"crypto/rand"
+	"encoding/json"
+	"fmt"
+
+	"github.com/btcsuite/btcutil/base58"
+	"golang.org/x/crypto/nacl/box"
+
+	"github.com/hyperledger/aries-framework-go/pkg/storage"
+)
+
+// BaseWallet wallet implementation
+type BaseWallet struct {
+	store storage.Store
+}
+
+// key json structure include public,private and metadata
+type key struct {
+	Public   string            `json:"public"`
+	Private  *[32]byte         `json:"private"`
+	Metadata map[string]string `json:"metadata"`
+}
+
+// New return new instance of wallet implementation
+func New(storeProvider storage.Provider) (*BaseWallet, error) {
+	store, err := storeProvider.GetStoreHandle()
+	if err != nil {
+		return nil, fmt.Errorf("failed to GetStoreHandle: %w", err)
+	}
+	return &BaseWallet{store: store}, nil
+}
+
+// CreateSigningKey create a new public/private signing keypair.
+func (w *BaseWallet) CreateSigningKey(metadata map[string]string) (*KeyInfo, error) {
```

Comment:
@llorllale @fqutishat is the signing algorithm fixed?

Reply:
yes

Reply:
@troyronda for signing what? DIDs and VCs? No.

Reply:
@fqutishat ^^^

Reply:
@fqutishat what will `metadata` be used for?

Reply:
maybe we should remove `Signing` from CreateSigningKey.. it's just a key

Signing will  follow https://w3c-dvcg.github.io/ld-signatures/ format that uses the raw key in the wallet to generate the JSON-LD signature.
Encryption for Pack/Unpack will use the raw key stored in the wallet to generate a JWE

Reply:
@Baha-sk @troyronda @llorllale https://github.com/hyperledger/aries-cloudagent-python/blob/master/aries_cloudagent/messaging/connections/manager.py#L154

Reply:
> @fqutishat what will `metadata` be used for?

it's driven from user

Reply:
I will remove metadata

Reply:
done